packages/sync-service/lib/electric/replication/shape_log_collector.ex

```diff
@@ -365,62 +396,73 @@ defmodule Electric.Replication.ShapeLogCollector do
-        )
+          OpenTelemetry.start_interval("unsubscribe_shape.remove_subscription")
 
-        {count, set}
-      end
+          OpenTelemetry.start_interval("unsubscribe_shape.remove_from_filter")
+          filter = Filter.remove_shape(state.filter, shape_handle)
 
-    OpenTelemetry.start_interval("unsubscribe_shape.remove_from_filter")
-    filter = Filter.remove_shape(state.filter, shape_handle)
+          OpenTelemetry.start_interval("unsubscribe_shape.remove_from_partitions")
+          partitions = Partitions.remove_shape(state.partitions, shape_handle)
 
-    OpenTelemetry.start_interval("unsubscribe_shape.remove_from_partitions")
-    partitions = Partitions.remove_shape(state.partitions, shape_handle)
+          OpenTelemetry.start_interval("unsubscribe_shape.remove_pids_by_shape_handle")
+          pids_by_shape_handle = Map.delete(state.pids_by_shape_handle, shape_handle)
 
-    OpenTelemetry.start_interval("unsubscribe_shape.remove_pids_by_shape_handle")
-    pids_by_shape_handle = Map.delete(state.pids_by_shape_handle, shape_handle)
+          OpenTelemetry.start_interval("unsubscribe_shape.remove_from_flush_tracker")
+          flush_tracker = FlushTracker.handle_shape_removed(state.flush_tracker, shape_handle)
 
-    OpenTelemetry.start_interval("unsubscribe_shape.remove_from_flush_tracker")
-    flush_tracker = FlushTracker.handle_shape_removed(state.flush_tracker, shape_handle)
+          OpenTelemetry.start_interval("unsubscribe_shape.remove_from_dependency_layers")
 
-    OpenTelemetry.start_interval("unsubscribe_shape.remove_from_dependency_layers")
-    dependency_layers = DependencyLayers.remove_dependency(state.dependency_layers, shape_handle)
+          dependency_layers =
+            DependencyLayers.remove_dependency(state.dependency_layers, shape_handle)
 
-    OpenTelemetry.stop_and_save_intervals(total_attribute: "unsubscribe_shape.total_duration_µs")
+          Electric.Shapes.ConsumerRegistry.remove_consumer(shape_handle, state.registry_state)
 
-    %{
-      state
-      | subscriptions: subscriptions,
-        filter: filter,
-        partitions: partitions,
-        pids_by_shape_handle: pids_by_shape_handle,
-        dependency_layers: dependency_layers,
-        flush_tracker: flush_tracker
-    }
-    |> log_subscription_status()
+          OpenTelemetry.stop_and_save_intervals(
+            total_attribute: "unsubscribe_shape.total_duration_µs"
+          )
+
+          {:ok,
+           %{
+             state
+             | subscriptions: count - 1,
+               filter: filter,
+               partitions: partitions,
+               pids_by_shape_handle: pids_by_shape_handle,
+               dependency_layers: dependency_layers,
+               flush_tracker: flush_tracker
+           }
+           |> log_subscription_status()}
+        else
+          Logger.warning("Received unsubscribe from unknown consumer: #{inspect(shape_handle)}")
```

Comment:
```suggestion
          Logger.error("Received unsubscribe from unknown consumer: #{inspect(shape_handle)}")
```
We were logging an error for this case before, which I think I've never seen on Sentry, so perhaps before downgrading it we should let it be an error if it's never supposed to happen?

extra nit: perhaps worth adding a span attribute to mark if this removal was unexpected/unknown if we want to keep this as a warning so we can maintain observability of these cases